runtime-api/bonded-currency/src/lib.rs

```diff
@@ -0,0 +1,78 @@
+// KILT Blockchain – https://botlabs.org
+// Copyright (C) 2019-2024 BOTLabs GmbH
+
+// The KILT Blockchain is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// The KILT Blockchain is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+// If you feel like getting in touch with us, you can do so at info@botlabs.org
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use parity_scale_codec::{alloc::string::String, Codec};
+use sp_std::vec::Vec;
+
+mod pool_details;
+pub use pool_details::*;
+
+sp_api::decl_runtime_apis! {
+	/// Runtime API to compute the collateral for a given amount and pool ID
+	/// and to query all pool IDs where the given account is the manager or owner.
+	#[api_version(1)]
+	pub trait BondedCurrency<Balance, PoolId, Operation, AccountId, BitType, PoolDetails, Error> where
```

Comment:
Just a personal opinion: the way I go with generics in runtime APIs is to use them for anything that is not locally defined. In this case, I see the `pool_details.rs` module has a bunch of types defined which are meant to be used in the API, so I would replace any generics with those concrete types here. You can't declare types in this crate and still not use them anywhere. The alternative is to declare those types in the runtime directly, but I prefer the API to make some choices about the types being returned, so I would simply re-assess the use of generics in here.

Reply:
I updated the api in: [f62d168](https://github.com/KILTprotocol/kilt-node/pull/828/commits/f62d1688bec11d4d1a5127bff412d23b21e89a3b)